File: VB/MapSample/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraMap
Imports System.IO

Namespace MapSample
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub


		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnSave.Click
			Dim dialog As New SaveFileDialog()
			dialog.DefaultExt = ".shp" ' Default file extension
			dialog.Filter = "Shape file (.shp)|*.shp" ' Filter files by extension

			If dialog.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
				Dim vl As VectorItemsLayer = TryCast(Me.mapControl1.Layers(0), VectorItemsLayer)
				If vl IsNot Nothing Then
					vl.ExportToShp(dialog.FileName, New ShpExportOptions() With {.ExportToDbf = True, .ShapeType = ShapeType.PolygonZ})
					MessageBox.Show("Shape File Saved")
					Return
				End If
				MessageBox.Show("An error occured")
			End If
		End Sub

		Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnShp.Click
			Dim dialog As New OpenFileDialog()
			dialog.DefaultExt = ".shp" ' Default file extension
			dialog.Filter = "Shape file (.shp)|*.shp" ' Filter files by extension

			If dialog.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
				shapefileDataAdapter1.FileUri = New Uri(dialog.FileName)
				Dim projectionPath As String = dialog.FileName.Substring(0, dialog.FileName.Length-3) & "prj"

				If File.Exists(projectionPath) Then
					btnSave.Enabled = True
					btnPrj.Text = "2. Projection Loaded automatically"
					btnPrj.Enabled = False

				Else
					btnPrj.Text = "2. Open Projection File (*.prj)"
					btnPrj.Enabled = True
					btnSave.Enabled = False
				End If
			End If
		End Sub

		Private Sub vectorItemsLayer1_DataLoaded(ByVal sender As Object, ByVal e As DataLoadedEventArgs) Handles vectorItemsLayer1.DataLoaded
				mapControl1.ZoomToFitLayerItems()
		End Sub

		Private Sub simpleButton3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnPrj.Click
			Dim dialog As New OpenFileDialog()
			dialog.DefaultExt = ".prj" ' Default file extension
			dialog.Filter = "Shape file (.prj)|*.prj" ' Filter files by extension

			If dialog.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
				shapefileDataAdapter1.SourceCoordinateSystem= ShapefileDataAdapter.LoadPrjFile(New Uri(dialog.FileName))
				btnSave.Enabled = True
			End If
		End Sub
	End Class
End Namespace

File: CS/MapSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraMap;
using System.IO;

namespace MapSample
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
        }


        private void simpleButton1_Click(object sender, EventArgs e)
        {
            SaveFileDialog dialog = new SaveFileDialog();
            dialog.DefaultExt = ".shp"; // Default file extension
            dialog.Filter = "Shape file (.shp)|*.shp"; // Filter files by extension

            if (dialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
                VectorItemsLayer vl = this.mapControl1.Layers[0] as VectorItemsLayer;
                if (vl != null)
                {
                    vl.ExportToShp(dialog.FileName, new ShpExportOptions() { ExportToDbf = true, ShapeType = ShapeType.PolygonZ });
                    MessageBox.Show("Shape File Saved");
                    return;
                }
                MessageBox.Show("An error occured");               
            }
        }

        private void simpleButton2_Click(object sender, EventArgs e)
        {
            OpenFileDialog dialog = new OpenFileDialog();
            dialog.DefaultExt = ".shp"; // Default file extension
            dialog.Filter = "Shape file (.shp)|*.shp"; // Filter files by extension

            if (dialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
                shapefileDataAdapter1.FileUri = new Uri(dialog.FileName);
                string projectionPath = dialog.FileName.Substring(0, dialog.FileName.Length-3) + "prj";

                if (File.Exists(projectionPath))
                {
                    btnSave.Enabled = true;
                    btnPrj.Text = "2. Projection Loaded automatically";
                    btnPrj.Enabled = false;

                }
                else
                {
                    btnPrj.Text = "2. Open Projection File (*.prj)";
                    btnPrj.Enabled = true;
                    btnSave.Enabled = false;
                }
            }
        }

        private void vectorItemsLayer1_DataLoaded(object sender, DataLoadedEventArgs e)
        {
                mapControl1.ZoomToFitLayerItems();
        }

        private void simpleButton3_Click(object sender, EventArgs e)
        {
            OpenFileDialog dialog = new OpenFileDialog();
            dialog.DefaultExt = ".prj"; // Default file extension
            dialog.Filter = "Shape file (.prj)|*.prj"; // Filter files by extension

            if (dialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
                shapefileDataAdapter1.SourceCoordinateSystem= ShapefileDataAdapter.LoadPrjFile(new Uri(dialog.FileName));
                btnSave.Enabled = true;
            }
        }
    }
}
